lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py

```diff
@@ -168,14 +168,17 @@ def ensure_tags(vpc_conn, resource_id, tags, add_only, check_mode):
             return {'changed': False, 'tags': cur_tags}
 
         to_delete = dict((k, cur_tags[k]) for k in cur_tags if k not in tags)
-        if to_delete and not add_only:
-            vpc_conn.delete_tags(resource_id, to_delete, dry_run=check_mode)
+        if to_delete and not add_only and not check_mode:
+            vpc_conn.delete_tags(resource_id, to_delete)
 
         to_add = dict((k, tags[k]) for k in tags if k not in cur_tags or cur_tags[k] != tags[k])
-        if to_add:
-            vpc_conn.create_tags(resource_id, to_add, dry_run=check_mode)
+        if to_add and not check_mode:
+            vpc_conn.create_tags(resource_id, to_add)
 
-        latest_tags = get_resource_tags(vpc_conn, resource_id)
+        if check_mode:
```

Comment:
Why's that? My reasoning for the if/else block is that if in check_mode, the resource won't have had its tags updated, so querying the resource's tags will provide incorrect info in check mode. However, out of check mode, it's worth validating what the resource actually thinks its tags are (we could just say that `latest_tags = tags` and not make the API call, but I'd rather be accurate, as that'll expose issues sooner)